src/main/java/com/battleship/backend/validators/PlaceShipsValidator.java

```diff
@@ -0,0 +1,50 @@
+package com.battleship.backend.validators;
+
+
+import com.battleship.backend.models.Boardable;
+import com.battleship.backend.models.PlaceRequest;
+import com.battleship.backend.models.Ship;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class PlaceShipsValidator implements Validator {
+
+    public PlaceShipsValidator(){}
+
+    public boolean isValid(Boardable board, PlaceRequest placeRequest) {
+        //check row within bounds, check col within bounds
+        //check ship does not exceed bounds
+        //Check space is not occupied
+        Ship ship = placeRequest.getShip();
+        int row = placeRequest.getRow();
+        int col = placeRequest.getCol();
+
+        if(ship == null || !ship.getClass().getName().equals("com.battleship.backend.models.Ship")){
+            return false;
+        }
```

Comment:
Why do you check the ship class?  placeRequest.getShip should only return a Ship object.

Reply:
It should, but it can also be null right? I'll delete the class check .